src/PIL/Image.py

```diff
@@ -3334,16 +3337,18 @@ def _open_core(
                 factory, accept = OPEN[i]
                 result = not accept or accept(prefix)
                 if isinstance(result, str):
-                    accept_warnings.append(result)
+                    warning_messages.append(result)
                 elif result:
                     fp.seek(0)
                     im = factory(fp, filename)
                     _decompression_bomb_check(im.size)
                     return im
-            except (SyntaxError, IndexError, TypeError, struct.error):
+            except (SyntaxError, IndexError, TypeError, struct.error) as e:
                 # Leave disabled by default, spams the logs with image
                 # opening failures that are entirely expected.
                 # logger.debug("", exc_info=True)
+                if warn_possible_formats:
+                    warning_messages.append(i + " opening failed. " + str(e))
```

Comment:
```suggestion
                if warn_possible_formats:
                    warning_messages.append(i + " opening failed. " + str(e))
```
The comment is probably no longer relevant with this change.